pkg/daemon/controller/controller.go

```diff
@@ -189,8 +189,12 @@ func (sc *ServiceController) reconcileService(service winsvc.Service, expected s
 	}
 
 	if !reflect.DeepEqual(config.Dependencies, expected.Dependencies) {
-		config.Dependencies = expected.Dependencies
-		updateRequired = true
+		// reflect.DeepEqual considers a nil slice not equal to an empty slice, so we need to include an extra check
+		// to ensure they both aren't of length 0, which is desired.
+		if !(len(config.Dependencies) == len(expected.Dependencies) && len(config.Dependencies) == 0) {
```

Comment:
Also, consider adding a unit test around this.

Reply:
+1

Reply:
done